File: backend/lib/L2_data/models/goals/remote_tracker.py
#  Copyright (c) 2022. Alexandr Moroz


from sqlalchemy import Column, ForeignKey, Integer, String
from sqlalchemy.orm import relationship

from ..base_model import BaseModel, TitleableUnique, WorkspaceBound


class RemoteTrackerType(TitleableUnique, BaseModel):
    pass


class RemoteTracker(WorkspaceBound, BaseModel):
    remote_tracker_type_id = Column(Integer, ForeignKey("remotetrackertypes.id"))
    type = relationship("RemoteTrackerType")
    url = Column(String, nullable=False)
    login_key = Column(String, nullable=False)
    password = Column(String)
    description = Column(String)

File: frontend/lib/L1_domain/api_schema/remote_tracker.dart
// Copyright (c) 2022. Alexandr Moroz

import 'base_upsert_schema.dart';

class RemoteTrackerUpsert extends BaseUpsert {
  RemoteTrackerUpsert({
    required int? id,
    required this.typeId,
    required this.url,
    required this.loginKey,
    required this.password,
    required this.description,
  }) : super(id: id);

  final int typeId;
  final String url;
  final String loginKey;
  final String password;
  final String? description;
}
